packages/dds/tree/src/test/edit-manager/editManger.spec.ts

```diff
@@ -0,0 +1,606 @@
+        });
+        manager.addSequencedChange({
+            sessionId: peerSessionId2,
+            seqNumber: 5,
+            refNumber: 0,
+            changeset: cs5,
+        });
+        manager.addSequencedChange({
+            sessionId: peerSessionId2,
+            seqNumber: 6,
+            refNumber: 0,
+            changeset: cs6,
+        });
+        checkChangeList(manager, rebaser);
+    });
+
+    it("Can rebase multiple interleaved peer and local changes", () => {
+        const { rebaser, manager } = editManagerFactory();
+        const cs1 = rebaser.mintChangeset(0);
+        const cs2 = rebaser.mintChangeset(cs1.outputContext);
+        const cs3 = rebaser.mintChangeset(cs2.outputContext);
+        const cs4 = rebaser.mintChangeset(0);
+        const cs5 = rebaser.mintChangeset(cs4.outputContext);
+        const cs6 = rebaser.mintChangeset(cs5.outputContext);
+        const cs7 = rebaser.mintChangeset(0);
+        manager.addLocalChange(cs7);
+        manager.addSequencedChange({
+            sessionId: peerSessionId1,
+            seqNumber: 1,
+            refNumber: 0,
+            changeset: cs1,
+        });
+        manager.addSequencedChange({
+            sessionId: peerSessionId2,
+            seqNumber: 2,
+            refNumber: 0,
+            changeset: cs4,
+        });
+        const cs8 = rebaser.mintChangeset(getLocalContext(manager));
+        manager.addLocalChange(cs8);
+        const cs9 = rebaser.mintChangeset(getLocalContext(manager));
+        manager.addLocalChange(cs9);
+        manager.addSequencedChange({
+            sessionId: localSessionId,
+            seqNumber: 3,
+            refNumber: 0,
+            changeset: cs7,
+        });
+        manager.addSequencedChange({
+            sessionId: peerSessionId1,
+            seqNumber: 4,
+            refNumber: 1,
+            changeset: cs2,
+        });
+        manager.addSequencedChange({
+            sessionId: peerSessionId1,
+            seqNumber: 5,
+            refNumber: 2,
+            changeset: cs3,
+        });
+        manager.addSequencedChange({
+            sessionId: localSessionId,
+            seqNumber: 6,
+            refNumber: 2,
+            changeset: cs8,
+        });
+        manager.addSequencedChange({
+            sessionId: peerSessionId2,
+            seqNumber: 7,
+            refNumber: 0,
+            changeset: cs5,
+        });
+        manager.addSequencedChange({
+            sessionId: localSessionId,
+            seqNumber: 8,
+            refNumber: 2,
+            changeset: cs9,
+        });
+        manager.addSequencedChange({
+            sessionId: peerSessionId2,
+            seqNumber: 9,
+            refNumber: 0,
+            changeset: cs6,
+        });
+        checkChangeList(manager, rebaser);
+    });
+
+    it("Can handle all possible interleaving of steps", () => {
+        const meta = {
+            clientData: makeArray(NUM_CLIENTS, () => ({ pulled: 0, numLocal: 0 })),
```

Comment:
```suggestion
            clientData: Array(NUM_CLIENTS).fill({ pulled: 0, numLocal: 0 }),
```

Reply:
I thought of that too, but they all get the same object if we do that. :)

Reply:
Good point!